pkg/cloud/cloud.go

```diff
@@ -821,27 +811,58 @@ func (c *cloud) ResizeOrModifyDisk(ctx context.Context, volumeID string, newSize
 	if err != nil {
 		return 0, err
 	}
-	needsModification, volumeSize, err := c.validateModifyVolume(ctx, volumeID, newSizeGiB, options)
+	request := &ec2.DescribeVolumesInput{
+		VolumeIds: []string{volumeID},
+	}
+	volume, err := c.getVolume(ctx, request)
+	if err != nil {
+		return 0, err
+	}
+	needsModification, volumeSize, err := c.validateModifyVolume(ctx, volumeID, newSizeGiB, options, *volume)
 
 	if err != nil || !needsModification {
 		return volumeSize, err
 	}
-
 	req := &ec2.ModifyVolumeInput{
 		VolumeId: aws.String(volumeID),
 	}
 	if newSizeBytes != 0 {
 		req.Size = aws.Int32(newSizeGiB)
 	}
-	if options.IOPS != 0 {
-		req.Iops = aws.Int32(options.IOPS)
-	}
 	if options.VolumeType != "" {
 		req.VolumeType = types.VolumeType(options.VolumeType)
+	} else {
+		req.VolumeType = volume.VolumeType
 	}
 	if options.Throughput != 0 {
 		req.Throughput = aws.Int32(options.Throughput)
 	}
+	maxIops, minIops, maxIopsPerGb, _, err := getVolumeLimits(string(req.VolumeType), options.Throughput)
+	if err != nil {
+		return 0, err
+	}
+
+	if options.IOPS != 0 {
+		sizeToUse := newSizeGiB
+		if req.Size != nil {
+			sizeToUse = *req.Size
+		}
+		req.Iops = aws.Int32(capIOPS(string(req.VolumeType), sizeToUse, options.IOPS, minIops, maxIops, maxIopsPerGb, false)) //TODO: See if we want to set allowIncrease to always true instead of false on modify?
```

Comment:
Do we need a tag for allowIncrease? I think we do to make the behavior consistent, or we could treat it as always true